src/BuildScriptGenerator/Node/NodeLanguageDetector.cs

```diff
@@ -123,31 +125,25 @@ public LanguageDetectorResult Detect(RepositoryContext context)
 
         private string DetectNodeVersion(dynamic packageJson)
         {
-            var nodeVersionRange = packageJson?.engines?.node?.Value as string;
-            if (nodeVersionRange == null)
+            var nodeVersion = packageJson?.engines?.node?.Value as string;
```

Comment:
Note that this field can be a specific version or a range. Node platform says 'I can generate a script if the engine version/range falls in the range that I support'. Node platform then generates a script and let's the underlying setup script to set the expected version.
### Scenario:
- Node platform says it supports `>=4 <13`
- User app's `engines` field says `13 || >=8 <13`
- Resulting range `>=8 <13` that is supplied to `benv`
- This will result in `1.2.13.0` being chosen as the version that is used.